build-now.lua

```diff
@@ -435,9 +435,18 @@ local function build_construction(bld)
     local construction_type = bld.type
     dfhack.buildings.deconstruct(bld)
 
+    -- check for and remove a construction on the tile already (i.e. building a construction on top of a wall)
+    local use_open_space_for_original_tile = true
+    for i, construction in ipairs(df.global.world.constructions) do
+        if pos.x == construction.pos.x and pos.y == construction.pos.y and pos.z == construction.pos.z then
+            df.global.world.constructions:erase(i)
```

Comment:
instead of erasing the item, we could instead call `utils.binsearch` in `create_and_link_construction` so we can reuse the entry.

Reply:
We should give the function two modes, then. Or maybe use two functions. If we give it two modes, what would the changed parameters be called?

Reply:
How about if we keep the signature of create_and_link_construction add it is (i.e. don't pass in the new bool) and instead rewrite create_and_link_construction like this:
1) check tile at pos to see if it's a construction
2) if so, use utils.binsearch to get a reference to the construction and overwrite it with our new configuration. The original tile value should already be set correctly.
3) if not, allocate a new construction like we do now.
4) write a new function called get_original_tile that does the check for what original tile should be for newly allocated constructions and call it from create_and_link_construction.

Reply:
scatterbrain programming time. this is an optimisation but people may be doing it on huge areas so it might be worth it